driver-core/src/main/com/mongodb/client/model/geojson/codecs/GeometryDecoderHelper.java

```diff
@@ -503,7 +448,11 @@ private static String decodeCoordinateReferenceSystemProperties(final BsonReader
     }
 
     private static void validateIsDocument(final BsonReader reader) {
-        if (reader.getCurrentBsonType() != BsonType.DOCUMENT) {
+        BsonType currentType = reader.getCurrentBsonType();
```

Comment:
Not a bug per se, when decoding a CoordinateReferenceSystem we jump straight to this method and we haven't read anything.